### ISSUE TITLE
Error on rooms create

### ISSUE THREAD

**Issue by lrojas89** [webkul/hotelcommerce#224]
In the admin view when I try to create a new room cannot activate the buttons and trow a error. In the system logs this is the error

Shop not found at line 404 in file classes/shop/Shop.php

**Comment by faizansari-bd** [comment]
Hi @lrojas89 
Can you please share the screenshots of the exact issue. Also please let us know if you have made any changes in the code or the database.

**Comment by raymond-theboy23** [comment]
Hi, I have the same problem, here is my screenshoot of the system.

https://drive.google.com/file/d/1Xcv8_rY4Lcjs1xTzwBJYE8qMQFEhgTkC/view

Thanks.

**Comment by lrojas89** [comment]
Hi @faizansari-bd 
I not modify the code, and the screenshoot of the system is the same of @raymond-theboy23 , also I tried with docker image webkul/qloapps_docker, I verified that php version in the image is PHP 7.2.33-1+ubuntu18.04.1 and the mysql is Ver 14.14 Distrib 5.7.31,  but the error is the same.

Inside the folder logs of the system one of the files have this message `Shop not found at line 404 in file classes/shop/Shop.php`
and when I activate in the config file defines.inc.php the _PS_MODE_DEV_ then I get this error `Unknown error on line 2151 in file /home/qloapps/www/hotelcommerce/classes/Tools.php
[8192] Function create_function() is deprecated` .

The screenshot are this:
https://drive.google.com/file/d/1fAU3fBYjcT1ntO5-VDwXvXX2c-_M_e5q/view?usp=sharing

https://drive.google.com/file/d/1pesN9laRz-S1w91Q0qRF2OgJzpUZSFdd/view?usp=sharing

**Comment by wdammak** [comment]
@lrojas89 the v1.5.x is not compatible with php 7.x only php5.x
Function create_function() is deprecated
fixed in develop version
https://github.com/webkul/hotelcommerce/blob/develop/classes/Tools.php
But in the end you can't continue on php7 anyway, you have to change the version of your php to 5.x

**Comment by lrojas89** [comment]
@wdammak I know that because I tested the version v1.5.0 with php v5.6.40 and php v7.3.22, but the issue is the same. If you read the olders comments you can see that  I tried with docker image **webkul/qloapps_docker** and I verified that php version in the image is PHP 7.2.33-1+ubuntu18.04.1 and the mysql is Ver 14.14 Distrib 5.7.31 (**the image is from docker hub)**, and the issue is the same.

**Comment by wdammak** [comment]
@lrojas89 download https://raw.githubusercontent.com/webkul/hotelcommerce/develop/classes/Tools.php
and test

**Comment by lrojas89** [comment]
@wdammak I test that you sugested and working, but I get the buttons activated 3 min later. In all of the system admin views the buttons activate fastly, but on the products view is slow, you know that caused this?

**Comment by sumitwebkul** [comment]
@lrojas89 

It is wrong to say that you can not run QloApps on PHP version 7.x.x. You can run QloApps on it. But while you set the developer mode on then you will get notices of deprecated functions but they are not removed from PHP version 7.x.x. If you set developer mode to false then notices will disappear.
So I recommend you disable developer mode and then try.

There may be some script that is taking time to finish execution or your server speed could the issue. If you have a public domain then I recommend you to create a ticket at https://webkul.uvdesk.com/en/customer/create-ticket/ and share your details there to resolve your issue Fastly.
Hope you understand.

Thanks & Regards.

**Comment by lrojas89** [comment]
@sumitwebkul First thanks for your interest in my issue, i test the qloapps in bare metal server in my home, i not have a public domain or the system hosted public for the internet acces, but testing i encountered the error in procedure `this.displayBulk()` in line 165 of the file product.js in `js/admin/products.js` the function `this.current_request = this.display(stack[0], false);` get a undefined value and because of that throw a timeout response. This avoid save buttons in the product view get activated.
If you can help me with that I will apreciated

**Comment by sumitwebkul** [comment]
Ok,

It is difficult to suggest a solution without checking the issue.

Can I know the below information? 
- PHP version you are using
- max_execution_timeout
- memory_limit
- Have you changed any code on your server?
- Have you enabled the debug mode on your server?
- If you know how to see ajax running on the page the send me the screenshot of those on the product page. So that we can know which tab is creating issues.

It may be the reason that execution on your server is very slow and timeout is occurring while executing scripts.

Waiting for your response.

Thanks & Regards.

**Comment by lrojas89** [comment]
@sumitwebkul This is the information about the implementation of the services.

Proxmox VM Resorces are 8Gb RAM, 4vCPU, 100Gb
SO: Alpine Linux 3.8
PHP: 5.6.40
Apache: 2.4.33
MariaDB: 10.2.5 
Qlo Apps version == 1.5.0

PHP MaxExecutionTimeout  == 360s
PHP MemoryLimit == 2048M
PHP Charset == UTF8
PHP Xdebug == Disable
Apache Timeout == 360
Apache Charset == UTF8

I not made any change in the sources code, the PS_MODE_DEV in app is disable, as you sugested.

The screenshots of the Firefox Debug Tools when I open the product form is that:
https://drive.google.com/file/d/1tpt_z3wJ9dGZBnPf57dRfz2a-46sWIHS/view?usp=sharing
https://drive.google.com/file/d/16RiqHpCDZD6MO2dk4vAmUlURkxd3I_JJ/view?usp=sharing

The tabs that have problems in the forms when updated the page, are:
Configuration, Occupancy, Booking and ModuleHotelreservationsystem

The other get Satus 200
Pack, Prices, SEO, Associations, Images and Features.

The screenshot of that are this:
https://drive.google.com/file/d/1x_BHMjIzM-y-tUu6td8qbhHAyfG1UwG8/view?usp=sharing
https://drive.google.com/file/d/1zpEilreyvAzPYP9zkUEBDsMHijvoaxuk/view?usp=sharing

**Comment by sumitwebkul** [comment]
@lrojas89 

It is difficult to find the exact issue from this data.
I recommend you to create a ticket at https://webkul.uvdesk.com/en/customer/create-ticket/ and list your issues there in the ticket.
We can check the issue by taking the remote of your system through any tool. It will be the fastest way to resolve the issue.
Hope you understand.

Thanks & Regards.

**Comment by wdammak** [comment]
> 
> 
> @wdammak I test that you sugested and working, but I get the buttons activated 3 min later. In all of the system admin views the buttons activate fastly, but on the products view is slow, you know that caused this?

For the slow it's because of the repetitive call of the Cart modules and those in the column on the right.
The team has been working on this bug and will improve performance in the next release, but I admit that there is still a lot to do!
I advised the team to correct all its warnings with version 7, because this will improve the performance of the application since between v7 and v5 there is at least a 40% gain in speed.

**Comment by lrojas89** [comment]
@wdammak  well I tested several version before detect the real problem that cause the issue related and found the problem in the max execution time configured in the ajax function of the file product.js.
When I trace the php function execution time, and the apache server scripts execution time only remain the javascript execution time, and in review of the file I found that the parameter is set to 30000 ms when change to 60000 ms everything work fine.
Thank you for the pacience with this issue.
